Jenkinsfile

```diff
@@ -0,0 +1,208 @@
+      }
+      when {
+        beforeAgent true
+        allOf {
+          expression { return params.intake_ci }
+          // expression { return env.ONLY_DOCS == "false" }
+        }
+      }
+      steps {
+        // withGithubNotify(context: 'Intake') {
+          deleteDir()
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            withGoEnv(){
+              sh(label: 'Run intake', script: './.ci/scripts/intake.sh')
+            }
+          }
+        }
+      // }
+    }
+    stage('Build and Test'){
+      failFast false
+      parallel {
+        /**
+        Build on a linux environment.
+        */
+        stage('linux build') {
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            allOf {
+              expression { return params.linux_ci }
+            }
+          }
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir(BASE_DIR){
+              withMageEnv(){
+                retry(2) { // Retry in case there are any errors to avoid temporary glitches
+                  sleep randomNumber(min: 5, max: 10)
+                  sh(label: 'Linux build', script: './.ci/scripts/build.sh')
+                }
+              }
+            }
+          }
+        }
+        /**
+          Run unit tests and report junit results.
+        */
+        stage('Unit Test') {
+          agent { label 'linux && immutable && debian-11' }
+          options { skipDefaultCheckout() }
+          environment {
+            PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+            HOME = "${env.WORKSPACE}"
+            TEST_COVERAGE = "true"
+          }
+          when {
+            beforeAgent true
+            allOf {
+              expression { return params.test_ci }
+            }
+          }
+          steps {
+              deleteDir()
+              unstash 'source'
+              dir("${BASE_DIR}"){
+                withMageEnv(){
+                  sh(label: 'Run Unit tests', script: './.ci/scripts/unit-test.sh')
+                }
+              }
+          }
+        }
+        /**
+        Packages Artifacts & Publishes release
+        */
+        stage('Package&Publish') {
+          agent { label 'linux && immutable && debian-11' }
+          options { skipDefaultCheckout() }
+          environment {
+            PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+            HOME = "${env.WORKSPACE}"
+            SNAPSHOT = "true"
+          }
+          when {
+            beforeAgent true
+            allOf {
+              expression { return params.release_ci }
+              anyOf {
+                branch 'main'
+                branch pattern: '\\d+\\.\\d+', comparator: 'REGEXP'
+                tag pattern: 'v\\d+\\.\\d+\\.\\d+.*', comparator: 'REGEXP'
+                expression { return params.Run_As_Main_Branch }
+              }
+            }
+          }
+          stages {
+            stage('Package') {
```

Comment:
on the other hand, this one should stay over our current CI package, right?

Reply:
The script & make command are a wrapper around the mage package  command.It's used to set conditions/vars before invoking mage.

Reply:
not sure this is answering what I asked

Reply:
If u mean the gh package ci job. Then no, it should stay in place. We aren't planning to run this workflow for pr checks and we want to know if 'mage package' fails before we merge a pr.

Reply:
alrighty